torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -449,6 +451,9 @@ def get_num_byte(dtype):
 
             # Step 1: Decide Kc assuming B block is L1-reside.
             size_cache_B = Kr * Kt_blocks * Nr * num_byte_B
+            if dtype_A is torch.bfloat16 and dtype_B is torch.int8 and Kr != 1:
+                # We will cache dequantized weights (BF16) in L1D for AMX micro-kernel
```

Comment:
Why does it have to be related to "AMX" here?

Reply:
For int8 WoQ GEMM, the AVX512 micro-kernel is only used when `M < Mr` for all AMX micro-kernel configs.
Weights are not reused in that case, so I only modified the AMX micro-kernel.

Reply:
Better to decouple the cache blocking from the implementation details of micro-kernel here. The comment on "AMX" seems an implementation detail that cache blocking had better not care much about.

Reply:
Thanks, but they actually seem to be coupled in case of int8 WoQ GEMM -

For AMX int8 WoQ GEMM micro-kernel, weights are converted from int8 -> BF16.
However, for AVX512, if I'd use a similar approach, then int8 weights would have to be converted to FP32.

Reply:
> I think I can't apply this suggestion because they are coupled in case of int8 WoQ GEMM -
> 
> For AMX int8 WoQ GEMM micro-kernel, weights are dequantized from int8 -> BF16. However, for AVX512, if I'd use a similar approach, then int8 weights would have to be dequantized to FP32.

Basically, it is a dequant to the compute type?